README.md

```diff
@@ -158,39 +157,59 @@ For full list of options please see [reference.conf](core/src/main/resources/ref
 
 ### Scala usage
 
-The Scala API is now _finally tagless_ (read more e.g. [here](https://www.beyondthelines.net/programming/introduction-to-tagless-final/)) -
-you can change the type it works with by specifying it when creating the _connection_. You have to provide`cats.arrow.FunctionK[Task, A]`
-and `cats.arrow.FunctionK[A, Task]` when creating new connection.
+The Scala API is now _finally tagless_ (read more e.g. [here](https://www.beyondthelines.net/programming/introduction-to-tagless-final/))
+meaning it can use whatever [`F[_]: cats.effect.Effect`](https://typelevel.org/cats-effect/typeclasses/effect.html)
+(e.g. `cats.effect.IO`, `monix.eval.Task`).
+Alternatively you are able to use any `F[_]` which is convertible to/from `monix.eval.Task` (see [Using own F](#using-own-non-effect-f))
 
 The Scala API uses types-conversions for both consumer and producer, that means you don't have to work directly with `Bytes` (however you
 still can, if you want) and you touch only your business class which is then (de)serialized using provided converter.
 
 The library uses two types of executors - one is for blocking (IO) operations and the second for callbacks. You _have to_ provide both of them:
 1. Blocking executor as `ExecutorService`
-1. Callback executor as `monix.execution.Scheduler` - you can get it e.g. by calling `Scheduler(myFavoriteExecutionContext)`
+1. Callback executor as `scala.concurrent.ExecutionContext`
 
 ```scala
 import com.typesafe.config.ConfigFactory
 import com.avast.metrics.api.Monitor
 import com.avast.clients.rabbitmq._ // for generic types support
 import com.avast.bytes.Bytes
-import monix.execution._
 import monix.eval._
 
 val config = ConfigFactory.load().getConfig("myRabbitConfig")
 
-implicit val sch: Scheduler = ???
+implicit val ec: ExecutionContext = ???
```

Comment:
Why is this actually needed?

Reply:
Because of the parts being implemented in `Task`. It's converted to `Scheduler` _inside_. The `Scheduler` is needed for conversion between `Task` and `F[_]: Effect`.